observation_portal/accounts/tasks.py

```diff
@@ -3,12 +3,31 @@
 from django.core.mail import send_mail as django_send_mail
 from django.core.mail import send_mass_mail as django_send_mass_mail
 from django.contrib.auth.models import User
+from django.conf import settings
 from oauth2_provider.models import AccessToken
+from urllib.parse import urljoin
 import logging
+import json
+import requests
+
 
 logger = logging.getLogger(__name__)
 
 
+@dramatiq.actor()
+def update_or_create_client_applications_user(user_json):
+    """
+    This call creates or updates the user account and profile information for each of the Oauth
+    Client applications specified in settings
+    """
+    for base_url in settings.OAUTH_CLIENT_APPS_BASE_URLS:
+        url = urljoin(base_url, '/authprofile/addupdateuser/')
+        logger.warning(f"Updating user details at {url}")
+        header = {'Authorization': f"Server {settings.OAUTH_SERVER_KEY}"}
+        response = requests.post(url, data=user_json, headers=header)
+        response.raise_for_status()
```

Comment:
This will raise an error on connection issues, which should just trigger dramatiq to retry with a exponential falloff.

Reply:
I suppose the only thing I can think of to optimize this is to fork out the tasks so that each one goes to a separate client app - e.g. moving the `for base_url in settings.OAUTH_CLIENT_APPS_BASE_URLS:` loop into the signal handlers, and spinning up N tasks, one for each client app.

Given that it's not a hugely expensive request/operation it's not a huge deal, but it does seem wasteful to retry and update all apps when only a subset of them fails.

Reply:
But maybe not...I guess I could see a race condition where a user revokes their token multiple times when, say, the science archive is down and not accepting connections. If there's multiple tasks running trying to update the science archive user who knows which one will hit first when it comes back online. May be better to update them all at once like you've done, so at least all client apps will be consistent.

Reply:
So I think I'll leave this as-is for now, since I'm not sure which approach will be better, and if we notice an issue with the authentication stuff getting out of sync due to this issue we can think about changing it later. On the one hand, I don't think this will normally be an issue, but say if the downtimeDB is taken down for a day or something I guess none of our user updates would propagate until it came back online. But probably in those cases we will furiously be trying to get back the service that is down anyway, so it shouldn't be an issue for long.